### DISCUSSION TITLE
Reduce 'Checking document' messages with ltex

### DISCUSSION THREAD

**Discussion by EricDriussi** [LazyVim/LazyVim#4299] [Q&A] [answered]
For some reason, when editing a file with ltex attatched it keeps showing these messages on every keypress.

![image](https://github.com/user-attachments/assets/e5f77943-b6bb-4cee-8d94-b3082e5fdf3e)

I understand that this is handled by noice.nvim, and that it probably has something to do with ltex-ls in particular (since other LSPs don't behave this way). But I can't seem to pin point the issue nor how to mitigate it.

I've tried fiddling around with the ltex-ls settings but nothing gives, I don't really have any particular config that might affect this (that I can think of at least).

```lua
{
  "folke/noice.nvim",
  opts = {
    cmdline = {
      view = "cmdline",
    },
    presets = {
      command_palette = false,
    },
    routes = {
      {
        filter = {
          event = "msg_show",
          kind = "",
          find = "written",
        },
        opts = { skip = true },
      },
    },
  },
},

{
  "neovim/nvim-lspconfig",
  opts = function(_, opts)
    local keys = require("lazyvim.plugins.lsp.keymaps").get()
    for _, key in ipairs(require("keys").lsp) do
      keys[#keys + 1] = key
    end

    opts.inlay_hints = { enabled = false }
    opts.servers.ansiblels = { filetypes = { "yaml", "yml" } }
    opts.diagnostics.virtual_text.severity = { min = vim.diagnostic.severity.ERROR }
    opts.diagnostics.virtual_text.prefix = diagn_icons.Error

    opts.diagnostics.float = {
      focusable = false,
      source = "if_many",
      header = "",
      prefix = "",
    }
  end,
},
```

Any clue on what might be causing this would help a lot, what the heck is ltex doing here?

**Comment by dpetka2001** [comment, marked as answer]
In `nvim-lspconfig` spec you could try
```lua
opts.servers = vim.tbl_deep_extend("force", opts.servers, {
  ltex = {
    settings = {
      ltex = {
        diagnosticSeverity = "error",
        checkFrequency = "save",
      },
    },
  },
})
```
One of the 2 should work I believe. Probably `checkFrequency` is the most relevant. Otherwise take a look at [ltex settings](https://valentjn.github.io/ltex/settings.html) in case it is something else.

**Reply by EricDriussi** [reply]
Spot on! I somehow missed the `checkFrequency` settings for the ltex lsp :facepalm: 

Thank you very very much!

**Comment by nhejazi** [comment]
sorry to bother, i'm relatively new to neovim and lazyvim and am trying to figure out how to adopt this solution. what would be the right way to include this in `nvim-lspconfig` without overriding too much else of what lazyvim configures? it seems like this naive approach disables the ltex server:

```lua
return {
  {
    "neovim/nvim-lspconfig",
    config = function(_, opts)
      -- modify LSP servers
      opts.servers = vim.tbl_deep_extend("force", opts.servers, {
        ltex = {
          settings = {
            ltex = {
              checkFrequency = "save",
            },
          },
        },
      })
    end,
  },
}
```

**Reply by dpetka2001** [reply]
Change `config` into `opts`.

**Reply by nhejazi** [reply]
awesome, thanks!

**Reply by julowe** [reply]
Thank you! I changed it for ltex_plus and added some markdown nodes, so sharing the entire file from my `~/.config/nvim/lua/plugins` dir in case it is useful. Verified it parses/works with `:checkhealth` in vim.

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = function(_, opts)
      -- modify LSP servers
      opts.servers = vim.tbl_deep_extend("force", opts.servers, {
        ltex_plus = {
          settings = {
            ltex = {
              checkFrequency = "save",
              markdown = {
                nodes = {
                  CodeBlock = "dummy",
                  FencedCodeBlock = "dummy",
                  AutoLink = "dummy",
                  Code = "dummy",
                },
              },
            },
          },
        },
      })
    end,
  },
}
```
